lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -64,7 +64,9 @@ sub open_connection {
         my ($c, $msg) = @_;
         # Incoming data will be JSON-formatted text, as a Unicode string.
         # We normalize the entire string before decoding.
-        my $args = eval { Mojo::JSON::decode_json(Unicode::Normalize::NFC($msg)) };
+        my $normalized_msg = Unicode::Normalize::NFC($msg);
+        my $args = eval { decode_json_utf8($normalized_msg) };
```

Comment:
I'd normally structure this as more of a try/catch block as in:

```
eval {
 my $args = decode...;
 on_message(...);
 1
} or do {
 log($@)
}
```

but I don't think we support (or expect much usage on) pre-5.14 Perl, and the decoding step failure modes aren't going to trigger the empty-`$@` issue so your current version is safe enough - the only remaining thing is that we're passing an `undef` value to `on_message()`, I'd be tempted to skip the call entirely.

However, if people are receiving invalid data here the current version does at least provide a mechanism to detect that and complain (e.g. drop the connection), so let's go with this as-is!

:1st_place_medal: